Tests/Tests/STPPaymentMethodsViewControllerTest.m

```diff
@@ -0,0 +1,237 @@
+@end
+
+@interface STPPaymentMethodsViewControllerTest : XCTestCase
+
+@end
+
+@implementation STPPaymentMethodsViewControllerTest
+
+- (STPPaymentMethodsViewController *)buildViewControllerWithCustomer:(STPCustomer *)customer
+                                                       configuration:(STPPaymentConfiguration *)config
+                                                            delegate:(id<STPPaymentMethodsViewControllerDelegate>)delegate {
+    STPTheme *theme = [STPTheme defaultTheme];
+    id<STPBackendAPIAdapter> mockAPIAdapter = [STPMocks staticAPIAdapterWithCustomer:customer];
+    STPPaymentMethodsViewController *vc = [[STPPaymentMethodsViewController alloc] initWithConfiguration:config
+                                                                                                   theme:theme
+                                                                                              apiAdapter:mockAPIAdapter
+                                                                                                delegate:delegate];
+    if (vc) {
+        XCTAssertNotNil(vc.view);
+    }
+    return vc;
+}
+
+/**
+ When the customer has no sources and the configuration doesn't use sources,
+ STPAddCardViewController should be shown.
+ */
+- (void)testInitWithNoSourcesAndUseSourcesOff {
+    STPCustomer *customer = [STPFixtures customerWithNoSources];
+    STPPaymentConfiguration *config = [STPFixtures paymentConfiguration];
+    config.useSourcesForCards = NO;
+    id<STPPaymentMethodsViewControllerDelegate>delegate = OCMProtocolMock(@protocol(STPPaymentMethodsViewControllerDelegate));
+    STPPaymentMethodsViewController *sut = [self buildViewControllerWithCustomer:customer
+                                                                   configuration:config
+                                                                        delegate:delegate];
+    XCTAssertTrue([sut.internalViewController isKindOfClass:[STPAddCardViewController class]]);
+}
+
+/**
+ When the customer has no sources and the configuration uses sources,
+ STPAddSourceViewController should be shown.
+ */
+- (void)testInitWithNoSourcesAndUseSourcesOn {
+    STPCustomer *customer = [STPFixtures customerWithNoSources];
+    STPPaymentConfiguration *config = [STPFixtures paymentConfiguration];
+    config.useSourcesForCards = YES;
+    id<STPPaymentMethodsViewControllerDelegate>delegate = OCMProtocolMock(@protocol(STPPaymentMethodsViewControllerDelegate));
+    STPPaymentMethodsViewController *sut = [self buildViewControllerWithCustomer:customer
+                                                                   configuration:config
+                                                                        delegate:delegate];
+    XCTAssertTrue([sut.internalViewController isKindOfClass:[STPAddSourceViewController class]]);
+}
+
+/**
+ When the customer has no sources and the configuration uses sources,
+ STPPaymentMethodsInternalVC should be shown.
```

Comment:
Right, but I mean the rest of the description is the same so something there is probably wrong? :)